core/deployment/src/main/java/io/quarkus/deployment/annotations/ExecutionTime.java

```diff
@@ -27,5 +27,10 @@
     /**
      * The bytecode is run from a main method
      */
-    RUNTIME_INIT
+    RUNTIME_INIT,
+
+    /**
+     * The bytecode is run from a main method after all RUNTIME_INIT
+     */
+    AFTER_STARTUP
```

Comment:
We don't want to use an entire separate bytecode recorder just for this one case.

Reply:
:+1:

Reply:
I just remembered another reason why I needed this recorder - besides needing to have a way to exit. Basically the command line entry points need to be run `after` everything in the else (I thinking basically of the Arc events here).
I didn't see a way to order the execution of runtime templates - did I miss something perhaps?

Reply:
Also, in the command line application case, it doesn't make much sense I think to show the startup time after the actual work that is intended has been performed.

So in my view the `AFTER_STARTUP` execution time makes sense for such applications. Is there a large performance penalty at build time of having another bytecode recorder?

Reply:
It's not a large penalty by itself, but we don't want to introduce a cost if we don't need to.  Since there can only ever be one of these (logically), it probably doesn't even need to be a bytecode recorder.  You could for example produce a build item that contains the method info (maybe just the descriptor) to call, and then just emit a call to that method at the right spot if the build item is present.

Reply:
Got it! I will look into it, thanks!

Reply:
I'll give this a go to ensure that it can be implemented without too much trouble but there is another problem that will arise if we don't have the extra bytecode recorder. Since we won't be able to capture arguments, we won't be able to support use cases like Aesh, were the user can write multiple commands and it would be our job to feed Aesh with the class names.
So in a non Aesh use case, yes there can only be one entry point (that makes absolute sense since it would be the equivalent of a main method) and no bytecode recorder is needed, but if we want to bring Aesh into the mix, not having one will probably be a problem. 
WDYT?

Reply:
I pushed a commit that removes the need for a new BytecodeRecorder. But it does mean that now some bytecode needs to be written by the extension :)

Reply:
We might be able to do something similar for Aesh